ecosystem_tests/dorkl.py

```diff
@@ -765,3 +794,385 @@ def basic_blueprint_test(blueprint_file_name,
         except Exception as e:
             logger.error('Error: {e}'.format(e=str(e)))
             cleanup_on_failure(test_name)
+
+
+def is_first_invocation(test_name):
+    """
+    Check if this is the first invocation of the test,
+    by check existence of blueprint and deployment with test_name id.
+    param: test_name: The test name.
+    """
+    logger.info(
+        'Checking if {test_name} in blueprints or deployments list '.format(
+            test_name=test_name))
+    blueprints_list = cloudify_exec('cfy blueprints list')
+    deployments_list = cloudify_exec('cfy deployments list')
+
+    def _map_func(bl_or_dep_dict):
+        return bl_or_dep_dict["id"]
+
+    if test_name in map(_map_func, blueprints_list) and test_name in map(
+            _map_func, deployments_list):
+        logger.info('Not first invocation!')
+        return False
+    elif test_name in map(_map_func, blueprints_list) or test_name in map(
+            _map_func, deployments_list):
+        raise EcosystemTestException(
+            'Found blueprint or deployment with name:{test_name}, please '
+            'delete the blueprint/deployment manually and rerun the '
+            'test'.format(
+                test_name=test_name))
+    else:
+        logger.info('First invocation!')
+        return True
+
+
+def validate_on_second_invoke_param(on_second_invoke=None):
+    if on_second_invoke and on_second_invoke not in ['resume', 'rerun',
+                                                     'update']:
+        raise EcosystemTestException('on_second_invoke param must be one of:'
+                                     ' resume, rerun, update')
+
+
+def basic_blueprint_test_dev(blueprint_file_name,
+                             test_name,
+                             inputs=None,
+                             timeout=None,
+                             on_second_invoke=None,
+                             on_failure=None,
+                             uninstall_on_success=True):
+    """
+    blueprint test.
+    :param blueprint_file_name:
+    :param test_name:
+    :param inputs:
+    :param timeout:
+    :param on_second_invoke: Should be one of: resume,rerun,update
+    :param on_failure  what should test do in failure.
+    Should be one of: None,rollback-full,rollback-partial,uninstall-force.
+    :param uninstall_on_success: Perform uninstall if the test succeeded,
+    and delete the test blueprint.
+    :return:
+    """
+    timeout = timeout or TIMEOUT
+    if inputs != '':
```

Comment:
I actually think we should get rid of `input=''` from the old test. Default should be `None`. Then we should check if it's a file like `inputs.yaml` or a dict `{'input1': 'value'}` or a string `-i input1=value`.